gql/builder.go

```diff
@@ -411,19 +420,15 @@ func ResolveListField(name string, parent string) graphql.FieldResolveFn {
 			return nil, err
 		}
 
-		if qr, ok := p.Context.Value(QueryReporterContextKey).(QueryReporter); ok && qr != nil {
+		if qr, ok := p.Context.Value(QueryFunctionReporterContextKey).(QueryFunctionReporter); ok && qr != nil {
```

Comment:
So following up on the same key idea, here you would just have the QueryReporterContextKey and then try to cast to a QueryFunctionReporter. This isn't a breaking change because implementations of the QueryReporter which don't also implement QueryFunctionReporter will do nothing here. However it works for adding the feature because any implementation that supports both interfaces will start working.